files/external_node_v2.rb

```diff
@@ -196,8 +196,10 @@ def enc(certname)
   end
   res = http.start { |http| http.request(req) }
 
-  raise "Error retrieving node #{certname}: #{res.class}\nCheck Foreman's /var/log/foreman/production.log for more information." unless res.code == "200"
-  res.body
+  return res.body if res.code == '200'
+
+  raise NodeRetrievalError, "Error retrieving node #{certname}: #{res.class}"\
+        "Check Foreman's /var/log/foreman/production.log for more information."
```

Comment:
Is this message displayed anywhere? If the node retrieval fails on the first run, then the user is only likely to see an error retrieving the cache file, which won't exist. If it fails on subsequent runs, the script will now only use the cache.